### ISSUE TITLE
Remove reliance on `meltano.core` services in manifest generation

### ISSUE THREAD

**Issue by WillDaSilva** [meltano/meltano#7270]
TL;DR: This will help address env var resolution bugs:
- https://github.com/meltano/meltano/issues/6991
- https://github.com/meltano/meltano/issues/7063
- https://github.com/meltano/meltano/issues/7023
- https://github.com/meltano/meltano/issues/7034

---

Currently manifest generation makes use of services from `meltano.core` including `ProjectSettingsService` and `PluginSettingsService`. This causes it to inherit some of the bugs, particularly around env vars, present in those services.

As the manifest aims to be the single source of truth from which the rest of Meltano can access settings and config values it will need to pull the flawed logic it's relying on from the `meltano.core` services into `meltano.core.manifest`, and fix it too.

At the same time, the `meltano.core` services can be refactored to use the manifest, thereby fixing the bugs in everything that relies on them.

In the future (beyond the scope of this issue) we may see fit to entirely remove some of these `meltano.core` services in favor of using the contexts provided by the manifest module.

Related:
- The Meltano settings Miro board: https://miro.com/app/board/uXjVO3j_ADw=/

**Comment by WillDaSilva** [comment]
Still relevant
